ra_aid/project_state.py

```
@@ -0,0 +1,66 @@
+        DirectoryAccessError: If the directory cannot be accessed
+        ProjectStateError: For other unexpected errors
+    """
+    try:
+        path = Path(directory).resolve()
+        if not path.exists():
+            raise DirectoryNotFoundError(f"Directory does not exist: {directory}")
+        if not path.is_dir():
+            raise DirectoryNotFoundError(f"Path is not a directory: {directory}")
+
+        # Get all files/dirs in the directory, excluding contents of .git
+        allowed_items: Set[str] = {'.git', '.gitignore'}
```

Ruff (0.8.2)

Local variable `allowed_items` is assigned to but never used

Remove assignment to unused variable `allowed_items`

(F841)